src/controllers/postController.ts

```diff
@@ -0,0 +1,69 @@
+// postController.ts
+import { Request, RequestHandler, Response } from "express";
+import { StatusCodes } from "http-status-codes";
+import { PostError } from "../errors/customErrors.ts";
+import { Post } from "../models/Post.ts";
+import {
+  buildPostDocument,
+  deletePostDocument,
+  updatePostDocument,
+} from "../services/postServices.ts";
+
+const createPostController = async (req: Request) => {
+  const postBody = req.body;
+  const createdPost = await buildPostDocument(postBody);
+  if (!createdPost) {
+    return null;
+  }
+  return { message: "Post created", data: createdPost };
+};
+
+const updatePostController = async (req: Request) => {
+  const postBody = req.body;
+  const { postid } = req.params;
+  const updatedPost = await updatePostDocument(postid, postBody);
+  if (!updatedPost) {
+    return null;
+  }
+  return {
+    message: "Post updated",
+    data: updatedPost,
+  };
+};
+
+const deletePostController = async (req: Request) => {
+  const { postid } = req.params;
```

Comment:
you should also check if any attachement of this post id with any other documents like User or Review. For example, you have not deleted all the review inside of this post array before deleting it. The Review array also has a subarray that has all the child review, also delete theme.